megamek/src/megamek/client/ui/swing/boardview/AbstractWreckSprite.java

```diff
@@ -70,12 +73,13 @@ public Rectangle getBounds() {
     public void prepare() {
         // figure out size
         String shortName = entity.getShortName();
-        Font font = new Font("SansSerif", Font.PLAIN, 10); //$NON-NLS-1$
+        Font font = new Font("SansSerif", Font.PLAIN, 10);
         Rectangle tempRect = new Rectangle(47, 55, bv.getFontMetrics(font)
                 .stringWidth(shortName) + 1, bv.getFontMetrics(font)
                 .getAscent());
 
         // create image for buffer
+        final Rectangle bounds = getBounds(); // more efficient than recreation
```

Comment:
First of all, why are we hiding the underlying class member variable?

Second, what this function call does is recreate the bounds from scratch (it invokes the AbstractWreckSprite implementation of getBounds()), so the comment is basically the exact opposite of what the code does.

I'm not saying it's a bad fix, but I think it needs a little elaboration, and the comment needs to be updated.

Reply:
I missed the word double here... because if we called getBounds it would recreate it twice.

I looked at using this getter before two super.getBounds(), this was the easier version to read.

Reply:
Ok, let's just avoid hiding the member variable then and update the comment.

Reply:
You could also do that.

Reply:
I was thinking update the local variable name, but either solution works.

Reply:
Although, I think your solution is probably better, since it forces an update on the class variable in case it's null or something dumb like that (which apparently it is in some cases).

Reply:
actually, it forces an update on the class variable either way, so never mind. Up to you.

Reply:
Updated the variable name and comment